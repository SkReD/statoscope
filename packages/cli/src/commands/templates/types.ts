export const enum FileExt {
  js = 'js',
  ts = 'ts',
}

export const enum ModuleType {
  commonjs = 'commonjs',
  esm = 'esm',
}

export type TemplateOptions =
  | {
      output: {
        fileExt: FileExt.js;
        module: ModuleType.commonjs | ModuleType.esm;
      };
    }
  | {
      output: {
        fileExt: FileExt.ts;
        module: ModuleType.esm;
      };
    };

export type Template<TAdditionalOptions = undefined> = (
  templateOptions: TemplateOptions,
  additionalOptions?: TAdditionalOptions,
) => string;

export type RuleTemplatesPartsFromFileExt = {
  [FileExt.ts]: {
    imports: string;
    body: string;
    exports: string;
  };
  [FileExt.js]: {
    imports: {
      esm: string;
      commonjs: string;
    };
    body: string;
    exports: {
      commonjs: string;
      esm: string;
    };
  };
};
