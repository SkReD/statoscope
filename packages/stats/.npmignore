*
!bin
!spec/**/*.{js,d.ts,json}
**/*.spec.{js,d.ts}
