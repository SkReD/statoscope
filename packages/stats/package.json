{
  "name": "@statoscope/stats",
  "version": "5.28.1",
  "description": "Statoscope stats spec",
  "types": "./spec/index.d.ts",
  "repository": "https://github.com/statoscope/statoscope/",
  "author": {
    "name": "Sergey Melyukov",
    "url": "https://twitter.com/smelukov"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "spec/*"
  ],
  "license": "MIT",
  "homepage": "https://github.com/statoscope/statoscope/"
}
