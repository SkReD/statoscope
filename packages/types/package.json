{
  "name": "@statoscope/types",
  "version": "5.28.1",
  "description": "Statoscope types",
  "types": "./types/index.d.ts",
  "repository": "https://github.com/statoscope/statoscope/",
  "author": {
    "name": "Sergey Melyukov",
    "url": "https://twitter.com/smelukov"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@statoscope/stats": "5.28.1"
  },
  "license": "MIT",
  "homepage": "https://github.com/statoscope/statoscope/"
}
