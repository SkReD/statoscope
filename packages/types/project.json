{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "types",
  "projectType": "library"
}
