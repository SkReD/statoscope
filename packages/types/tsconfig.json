{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "dist"
  },
  "exclude": ["./dist", "**/*.spec.ts"],
  "references": [
    {"path": "../stats"}
  ]
}
