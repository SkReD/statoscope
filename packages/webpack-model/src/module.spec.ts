import stats from '../../../test/bundles/v5/simple/stats-prod.json';
import { NormalizedModule, NormalizedReason } from '../types';
import { moduleReasonResource, moduleResource, nodeModule } from './module';

describe('moduleResource', () => {
  test('should work', () => {
    const resource = [];

    expect(moduleResource(null)).toBeNull();

    for (const module of stats.modules as unknown as NormalizedModule[]) {
      resource.push({
        before: module.name,
        after: moduleResource(module),
      });
    }

    expect(resource).toMatchSnapshot();
  });
});

describe('moduleReasonResource', () => {
  test('should work', () => {
    const reasons = [];

    expect(moduleReasonResource(null)).toBeNull();

    for (const module of stats.modules as unknown as NormalizedModule[]) {
      for (const reason of module.reasons) {
        reasons.push({
          before: reason.moduleName,
          after: moduleReasonResource(reason),
        });
      }
    }

    for (const chunk of stats.chunks) {
      for (const reason of chunk.origins as unknown as NormalizedReason[]) {
        reasons.push({
          before: reason.moduleName,
          after: moduleReasonResource(reason),
        });
      }
    }

    expect(reasons).toMatchSnapshot();
  });
});

describe('nodeModule', () => {
  test('should work', () => {
    expect(nodeModule(null)).toBeNull();

    expect(nodeModule('./node_modules/foo/src')).toMatchSnapshot();
    expect(nodeModule('./node_modules/@foo/bar/src')).toMatchSnapshot();

    expect(nodeModule('./node_modules/foo/node_modules/bar/src')).toMatchSnapshot();
    expect(nodeModule('./node_modules/foo/node_modules/@bar/baz/src')).toMatchSnapshot();

    expect(
      nodeModule('./node_modules/@foo/bar/node_modules/@baz/quux/src'),
    ).toMatchSnapshot();
  });
});
