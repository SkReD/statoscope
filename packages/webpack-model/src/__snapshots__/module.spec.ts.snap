// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moduleReasonResource should work 1`] = `
[
  {
    "after": null,
    "before": null,
  },
  {
    "after": null,
    "before": null,
  },
  {
    "after": "simple/src/index.ts",
    "before": "./simple/src/index.ts + 3 modules",
  },
  {
    "after": "simple/src/index.ts",
    "before": "./simple/src/index.ts + 3 modules",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "node_modules/is-array/index.js",
    "before": "./node_modules/is-array/index.js",
  },
  {
    "after": "simple/src/index.ts",
    "before": "./simple/src/index.ts",
  },
  {
    "after": "simple/src/index.ts",
    "before": "./simple/src/index.ts",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "simple/src/sub-module.ts",
    "before": "./simple/src/sub-module.ts",
  },
  {
    "after": "simple/src/sub-module.ts",
    "before": "./simple/src/sub-module.ts",
  },
  {
    "after": "simple/src/sub-module.ts",
    "before": "./simple/src/sub-module.ts",
  },
  {
    "after": "simple/src/sub-module.ts",
    "before": "./simple/src/sub-module.ts",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "node_modules/css-loader/dist/runtime/api.js",
    "before": "./node_modules/css-loader/dist/runtime/api.js",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "node_modules/css-loader/dist/runtime/getUrl.js",
    "before": "./node_modules/css-loader/dist/runtime/getUrl.js",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "simple/src/index.ts",
    "before": "./simple/src/index.ts + 3 modules",
  },
  {
    "after": "simple/node_modules/foo/index.js",
    "before": "./simple/node_modules/foo/index.js",
  },
  {
    "after": null,
    "before": "",
  },
  {
    "after": null,
    "before": "",
  },
  {
    "after": null,
    "before": "",
  },
  {
    "after": null,
    "before": "",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts",
  },
  {
    "after": null,
    "before": "",
  },
  {
    "after": null,
    "before": "",
  },
]
`;

exports[`moduleResource should work 1`] = `
[
  {
    "after": "simple/src/index.ts",
    "before": "./simple/src/index.ts + 3 modules",
  },
  {
    "after": "simple/src/index2.ts",
    "before": "./simple/src/index2.ts + 1 modules",
  },
  {
    "after": "node_modules/is-array/index.js",
    "before": "./node_modules/is-array/index.js",
  },
  {
    "after": "simple/src/sub-module.ts",
    "before": "./simple/src/sub-module.ts",
  },
  {
    "after": "simple/src/index2.css",
    "before": "./simple/src/index2.css",
  },
  {
    "after": "external "extLib"",
    "before": "external "extLib"",
  },
  {
    "after": "simple/src/test.txt",
    "before": "./simple/src/test.txt",
  },
  {
    "after": "simple/src/dyn.ts",
    "before": "./simple/src/dyn.ts",
  },
  {
    "after": "simple/node_modules/foo/index.js",
    "before": "./simple/node_modules/foo/index.js",
  },
  {
    "after": "simple/node_modules/bar/index.js",
    "before": "./simple/node_modules/bar/index.js",
  },
  {
    "after": "node_modules/css-loader/dist/runtime/api.js",
    "before": "./node_modules/css-loader/dist/runtime/api.js",
  },
  {
    "after": "node_modules/css-loader/dist/runtime/getUrl.js",
    "before": "./node_modules/css-loader/dist/runtime/getUrl.js",
  },
  {
    "after": "simple/src/statoscope.png",
    "before": "./simple/src/statoscope.png",
  },
  {
    "after": "simple/node_modules/foo/node_modules/bar/index.js",
    "before": "./simple/node_modules/foo/node_modules/bar/index.js",
  },
  {
    "after": "webpack/runtime/chunk loaded",
    "before": "webpack/runtime/chunk loaded",
  },
  {
    "after": "webpack/runtime/compat get default export",
    "before": "webpack/runtime/compat get default export",
  },
  {
    "after": "webpack/runtime/define property getters",
    "before": "webpack/runtime/define property getters",
  },
  {
    "after": "webpack/runtime/ensure chunk",
    "before": "webpack/runtime/ensure chunk",
  },
  {
    "after": "webpack/runtime/get javascript chunk filename",
    "before": "webpack/runtime/get javascript chunk filename",
  },
  {
    "after": "webpack/runtime/global",
    "before": "webpack/runtime/global",
  },
  {
    "after": "webpack/runtime/hasOwnProperty shorthand",
    "before": "webpack/runtime/hasOwnProperty shorthand",
  },
  {
    "after": "webpack/runtime/jsonp chunk loading",
    "before": "webpack/runtime/jsonp chunk loading",
  },
  {
    "after": "webpack/runtime/load script",
    "before": "webpack/runtime/load script",
  },
  {
    "after": "webpack/runtime/make namespace object",
    "before": "webpack/runtime/make namespace object",
  },
  {
    "after": "webpack/runtime/publicPath",
    "before": "webpack/runtime/publicPath",
  },
]
`;

exports[`nodeModule should work 1`] = `
{
  "isRoot": true,
  "name": "foo",
  "path": "./node_modules/foo",
}
`;

exports[`nodeModule should work 2`] = `
{
  "isRoot": true,
  "name": "@foo/bar",
  "path": "./node_modules/@foo/bar",
}
`;

exports[`nodeModule should work 3`] = `
{
  "isRoot": false,
  "name": "bar",
  "path": "./node_modules/foo/node_modules/bar",
}
`;

exports[`nodeModule should work 4`] = `
{
  "isRoot": false,
  "name": "@bar/baz",
  "path": "./node_modules/foo/node_modules/@bar/baz",
}
`;

exports[`nodeModule should work 5`] = `
{
  "isRoot": false,
  "name": "@baz/quux",
  "path": "./node_modules/@foo/bar/node_modules/@baz/quux",
}
`;
